testworkload/nginx_workload.go

```diff
@@ -119,8 +120,9 @@ func (n *Nginx) IsReachable() error {
 
 // IsUnReachable pings the nginx service.
 // Expects the nginx service to be unreachable.
-func (n *Nginx) IsUnReachable() error {
-	if err := n.newPingPod(false); err != nil {
+// labels can be used to identify the pods that the job will spin up.
+func (n *Nginx) IsUnReachable(labels map[string]string) error {
```

Comment:
this api is really confusing.

lables should not be the opition of IsUnReachable. If you want to provide the flexibility of changing the job spec, we should somehow expose that directly rather than adding random arg to IsUnReachable.